lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -145,7 +145,15 @@ def compile_alias(tree, name, constant)
         def compile_object(tree, name, value)
           return if symbol_ignored?(name)
           klass = class_of(value)
-          klass_name = name_of(klass)
+
+          klass_name = if klass == ObjectSpace::WeakMap
+            # WeakMap is an implicit generic with one type variable
+            "ObjectSpace::WeakMap[T.untyped]"
```

Comment:
Why can't we treat this one like the other classes extending `T::Generic`? https://github.com/sorbet/sorbet/blob/master/rbi/stdlib/objspace.rbi#L170-L171

Reply:
As you point out, the fact that this is generic is encoded in the RBI file, but no artifact of that exists at runtime.

Reply:
I wonder if such a list should exist in `sorbet-runtime` so it can "tag" known generic types?

Reply:
TBH, there aren't that many of them, and, IMO, `ObjectSpace::WeakMap` shouldn't be marked as a generic type anyway. I'll see what upstream thinks about making it non-generic.

Reply:
I think `sorbet-runtime` could leave some info in the instance about this so we can pick it up from `tapioca`